src/Planner/PlannerActionsVisitor.cpp

```diff
@@ -159,7 +159,17 @@ class ActionNodeNameHelper
                 const auto & function_node = node->as<FunctionNode &>();
                 if (function_node.getFunctionName() == "__actionName")
                 {
-                    result = toString(function_node.getArguments().getNodes().at(1)->as<ConstantNode>()->getValue());
+                    /// Perform sanity check, because user may call this function with unexpected arguments
```

Comment:
This function should be allowed to be called by user. This is a implementation specific function.

Reply:
How can we forbid it? One possible option to have a specific setting in context that allows creating such functions, but it become more tricky to distinguish cases when query comes from remote server. Probably, `__actionName` cannot be transferred to remote, but I'm not sure (maybe some complex case with `Merge` over `Distributed`).

Reply:
We have ClientInfo in the context check if query come from user or from another server. The initiator should throw an exception if user calls this function.

Reply:
Ok, let's merge this PR and backport since it's a trivial one and prevents segfault, and then discuss something more sophisticated to forbid `__actionName` call from SQL in more